processing/src/main/java/io/druid/segment/StringDimensionHandler.java

```diff
@@ -120,24 +120,20 @@ public void validateSortedEncodedArrays(
               dimValName
           );
         }
+      } else {
```

Comment:
@leventov @jisookim0513 

re: `lhs.length = 2 and rhs.length = 0`, for this case, it should throw an exception at the `lhsLen != rhsLen` check before entering the loop, the StringDimensionHandler patch misplaced the `else` block after the `lhsLen == 0 && rhsLen == 1` checks

For this example:
`Map<String, Object> event1 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList("", ""),
        "predators", Arrays.asList("seal")
    );
    Map<String, Object> event2 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList(""),
        "predators", Arrays.asList("seal")
    );
`

The "penguins" rows should be treated differently I think, a row having two values instead of one, even if null, would affect the results in GroupBys which split the multivalued rows. 

Based on what @gianm told me, the equivalence between empty row and single-valued row there is related to how a row with a null value can be represented differently based on whether a segment is multivalue or singlevalue (empty row or single 0 value, respectively), but there isn't a wider equivalence between N nulls and 0/1 nulls

Having a `lhsLen != rhsLen` check before entering the loop seems useful/efficient, so I think the `(lhsLen == 0 && rhsLen == 1)` and `else if (rhsLen == 0 && lhsLen == 1)` checks are still needed in that case